src/main/java/org/owasp/benchmark/score/BenchmarkScore.java

```diff
@@ -638,221 +645,249 @@ private static TestResults readActualResults(File fileToParse) throws Exception
+					tr = new AppScanDynamicReader().parse( root );
+				}
 
-            } // end else
-         } // end if endsWith ".xml"
+				else if ( nodeName.equals( "xml-report" ) ) {
+					// For Appscan, this node has name="AppScan Report" and technology="SAST" or "DAST"
+					String name = Reader.getAttributeValue( "name", root );
+					if ("AppScan Report".equals(name)) {
+						String tech = Reader.getAttributeValue( "technology", root );
+						if ("SAST".equals(tech)) {
+							tr = new AppScanSourceReader2().parse( fileToParse );
+						} else if ("DAST".equals(tech)) {
+							tr = new AppScanDynamicReader2().parse( fileToParse );
+						} else System.out.println("Found AppScan Report with unfamiliar technology type: " + tech);
+					} else System.out.println("Found xml-report that was expected to have a name 'AppScan Report "
+								+ " but had name: " + name);
+				}
+
+				else if ( nodeName.equals( "issues" ) ) {
+					tr = new BurpReader().parse( root );
+				}
+
+				else if ( nodeName.equals( "netsparker" ) ) {
+					tr = new NetsparkerReader().parse( root );
+				}
+
+				else if ( nodeName.equals( "noisycricket" ) ) {
+					tr = new NoisyCricketReader().parse( root );
+				}
+
+				else if ( nodeName.equals( "VulnSummary" ) ) {
+					tr = new Rapid7Reader().parse( root );
+				}
+
+				else if ( nodeName.equals( "Report" ) ) {
+					tr = new SnappyTickReader().parse( root );
+				}
+
+				else if ( nodeName.equals( "Scan" ) ) {
+					tr = new WebInspectReader().parse( root );
+				}
+
+				else if ( nodeName.equals( "WAS_SCAN_REPORT" ) ) {
+					tr = new QualysWASReader().parse( root );
+				}
+
+				else System.out.println("Error: No matching parser found for XML file: " + filename);
+
+			} // end else
+		 } // end if endsWith ".xml"
 
 		else if ( filename.endsWith( ".fpr" ) ) {
 
 		// .fpr files are really .zip files. So we have to extract the .fvdl file out of it to process it
-		    Path path = Paths.get(fileToParse.getPath());
-		    FileSystem fileSystem = FileSystems.newFileSystem(path, null);
-		    File outputFile = File.createTempFile( filename, ".fvdl");
-		    Path source = fileSystem.getPath("audit.fvdl");
-		    Files.copy(source, outputFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
-		    tr = FortifyReader.parse( outputFile );
-		    outputFile.delete();
-
-		    // Check to see if the results are regular Fortify or Fortify OnDemand results
-		    // To check, you have to look at the filtertemplate.xml file inside the .fpr archive
-		    // and see if that file contains: "Fortify-FOD-Template"
-		    outputFile = File.createTempFile( filename + "-filtertemplate", ".xml");
-		    source = fileSystem.getPath("filtertemplate.xml");
-
-		    // In older versions of Fortify, like 4.1, the filtertemplate.xml file doesn't exist
-		    // So only check it if it exists
+			Path path = Paths.get(fileToParse.getPath());
+			FileSystem fileSystem = FileSystems.newFileSystem(path, (java.lang.ClassLoader) null);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fileSystem*. Currently, there are execution paths that do not contain closure statements. Either a) close *fileSystem* in a try-finally block or b) close the resource by declaring *fileSystem* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).